TCAT/Controllers/HomeOptionsCardViewController.swift

```diff
@@ -0,0 +1,561 @@
+                recentLocations = Array(recentLocations.prefix(2))
+            }
+            if !isNetworkDown {
+                sections = createSections()
+            }
+        }
+    }
+    var favorites: [Place] = [] {
+        didSet {
+            if favorites.count > maxFavoritesCount {
+                favorites = Array(favorites.prefix(2))
+            }
+            if !isNetworkDown {
+                sections = createSections()
+            }
+        }
+    }
+    var sections: [Section] = [] {
+        didSet {
+            tableView.reloadData()
+            delegate?.updateSize()
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupTableView()
+        setupInfoButton()
+        setupSearchBarSeparator()
+        setupSearchBar()
+        
+        setupConstraints()
+        
+        updatePlaces()
+    }
+
+    func setupTableView() {
+        tableView = UITableView(frame: .zero, style: .grouped)
+        tableView.backgroundColor = view.backgroundColor
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.separatorStyle = .none
+        tableView.keyboardDismissMode = .onDrag
+        tableView.tableFooterView = UIView(frame: .zero)
+        tableView.showsVerticalScrollIndicator = false
+        tableView.rowHeight = tableViewRowHeight
+        tableView.register(PlaceTableViewCell.self, forCellReuseIdentifier: Constants.Cells.placeIdentifier)
+        tableView.register(GeneralTableViewCell.self, forCellReuseIdentifier: Constants.Cells.seeAllStopsIdentifier)
+        view.addSubview(tableView)
+    }
+    
+    func setupSearchBar() {
+        searchBar = UISearchBar()
+        searchBar.placeholder = Constants.General.searchPlaceholder
+        searchBar.delegate = self
+        searchBar.searchBarStyle = .default
+        let textFieldInsideSearchBar = searchBar.value(forKey: "searchField") as? UITextField
+        textFieldInsideSearchBar?.backgroundColor = .white
+        searchBar.barTintColor = .white
+        searchBar.layer.borderColor = UIColor.white.cgColor
+        searchBar.layer.borderWidth = 1
+        view.addSubview(searchBar)
+    }
+    
+    func setupInfoButton() {
+        infoButton.imageEdgeInsets = UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 0)
+        infoButton.contentVerticalAlignment = .center
+        infoButton.tintColor = .black
+        infoButton.addTarget(self, action: #selector(openInformationScreen), for: .touchUpInside)
+        view.addSubview(infoButton)
+    }
+    
+    func setupSearchBarSeparator() {
+        searchBarSeparator = UIView()
+        searchBarSeparator.backgroundColor = Colors.backgroundWash
+        view.addSubview(searchBarSeparator)
+    }
+    
+    func setupConstraints() {
+        infoButton.snp.makeConstraints { (make) in
+            make.centerY.equalTo(searchBar)
+            make.trailing.equalToSuperview().inset(16)
```

Comment:
Inconsistent constant usage (variables vs. numbers). Just an FYI, you can honestly leave it